lmdeploy/pytorch/engine/engine.py

```diff
@@ -261,6 +262,20 @@ def _response(self,
                      data=data,
                      err_msg=err_msg))
 
+    def _get_max_session_len(self):
+        """get max session len."""
+        session_len = self.scheduler_config.max_session_len
+        max_tokens = (self.cache_config.num_gpu_blocks *
+                      self.cache_config.block_size)
+        window_size = self.cache_config.window_size
+        if window_size > 0 and window_size <= max_tokens:
```

Comment:
what's the `window_size`?

Reply:
sliding window attention

Reply:
Suppose `window_size == 2048` and `session_len is None`
According to the logic, `session_len = max_tokens = (1 << 63) - 1`
Is it expected?

Reply:
Does it mean that sliding window attention supports enumerous session length?

Reply:
> Does it mean that sliding window attention supports infinite session length?
Yes